chia/wallet/wallet_blockchain.py

```diff
@@ -57,30 +54,20 @@ async def create(
         self._latest_timestamp = uint64(0)
         self._height_to_hash = {}
         self._block_records = {}
-        if self.synced_weight_proof is not None:
-            await self.new_weight_proof(self.synced_weight_proof)
-        else:
-            self._sub_slot_iters = constants.SUB_SLOT_ITERS_STARTING
-            self._difficulty = constants.DIFFICULTY_STARTING
+        self._sub_slot_iters = constants.SUB_SLOT_ITERS_STARTING
+        self._difficulty = constants.DIFFICULTY_STARTING
 
         return self
 
-    async def new_weight_proof(self, weight_proof: WeightProof, records: Optional[List[BlockRecord]] = None) -> None:
+    async def new_valid_weight_proof(self, weight_proof: WeightProof, records: List[BlockRecord]) -> None:
```

Comment:
we where passing already validated WP's here since we already validate in "fetch_and_validate"